src/bsp/src/posix/src/BSP.cpp

```diff
@@ -50,7 +54,11 @@ void BSP::initChip(void* args) {
 
     m_rosNodeHandle = std::shared_ptr<ros::NodeHandle>(new ros::NodeHandle("~"));
 
-    xTaskCreate(rosWatcher, "ros_watch", configMINIMAL_STACK_SIZE, NULL, tskIDLE_PRIORITY + 1,
+    TCPUartMock& tcpUart = static_cast<TCPUartMock&>(BSPContainer::getHostUart());
+    int port = m_rosNodeHandle->param("uart_mock_port", 0);
+    tcpUart.openSocket(port);
```

Comment:
Would this belong in the container? 
You can get the node handle from the bsp in the container

You could use an [std::call_once](https://en.cppreference.com/w/cpp/thread/call_once) just after the static variable declaration

Reply:
This is the only place where I've managed to get it working every time. If I put it in the container, it doesn't always work. Also, I think it makes sense that the socket is opened as part of the init chip routine, it's part of the initialisation process. Imagine if we don't do a get() on the driver, the socket isn't opened and the phone can't connect.